### PR TITLE
19.0 code review

### PR DESCRIPTION
**Pull request by vertiond**
Vertcoin-Core [19.0-rc1 is a rebase](https://github.com/vertcoin-project/vertcoin-core/tree/19.0) on upstream Bitcoin-Core [22.x](https://github.com/bitcoin/bitcoin/tree/22.x)

### CODE REVIEW

**Comment by chadouming** [comment]
Been testing this for a while. So far, all the task I do with vertcoind/vertcoin-cli works fine. I'm creating block fine while mining and I can list/explore transaction/mempool. 

+1 from me

**Review by cruelnovo — COMMENTED** [review]
## In prior release but not in new release (non-critical) ##
contrib/devtools/gen-manpages.sh # Can we generate Vertcoin branded manpages?

contrib/seeds/makeseeds.py # Slight differences between v0.18 & v19

src/test/netbase_tests.cpp  # Looks like the ports were not updated for Vertcoin

src/qt/bitcoinstrings.cpp  # `*vertcoin_strings[]` vs `*bittcoin_strings[]` - 0.18 has it as `*bittcoin_strings[]` - also could use a QT_TRANSLATE_NOOP("vertcoin-core", "Vertcoin Core") in the file

src/Makefile.am # How come bignum.h was removed?  Line 700, (vertcoin_bin_ldadd) should be used as it was in v0.18?

## Vertcoin rebranding strings ##
doc/man/ # Since we're renaming all the manpage files to 'vertcoin-X.1', should we also update the contents to be vertcoin branded?

src/qt/guiutil.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

src/qt/intro.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

src/qt/sendcoinsdialog.cpp  # Missed some Vertcoin branding and PSVT vs PSBT crops up again

src/qt/forms/receiverequestdialog.ui # Missed an update for vtc1 and VTC string

src/qt/bitcoingui.cpp  # PSBT vs PSVT - 0.18 has it PSVT (Partially Sent Vertcoin Transaction)

src/rpc/misc.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

src/wallet/test/wallet_tests.cpp  # Update strings from bitcoin to vertcoin

src/bitcoin-cli.cpp # Update for a number of strings from bitcoin to vertcoin

src/bitcoind.cpp # Update for a number of strings from bitcoin to vertcoin

## Vertcoin not updated in comments (but was updated in prior release) ##
src/util/system.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments.  How is the CopyrightHolders function working? Why the strPrefix2?

src/policy/fees.cpp # only a difference in a comment. BTC-per-kb vs VTC-per-kb

src/Makefile.qt.include # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

## General questions ##
src/util/system.h # How is the CopyrightHolders function working? Why the strPrefix2?

src/crypto/verthash.cpp # How did you know to use `gArgs.GetDataDirNet()` instead of `GetDataDir()` (0.18 uses `GetDataDir()`)

src/crypto/verthash_datafile.cpp # How did you know to use `gArgs.GetDataDirNet()` instead of `GetDataDir()` (0.18 uses `GetDataDir()`)

contrib/seeds/nodes_test.txt  # can we provide a list of known testnet seeds? or remove this file?

contrib/verifybinaries/ # Should we remove this directory from Vertcoin?

src/bignum.h # is this part of KimotoGravityWell?  It is used by pow.cpp

src/chainparamsseeds.h # How come the seeds are not needed for Vertcoin in BIP155 format?

src/pow.cpp # How did you know to fix the "blockstogoback" logic (line 87-94)

configure.ac  # how come we're no longer enabling the SSE2 compiler extension?

src/chainparams.cpp # Should we set nPowTargetTimespan to 3.5 days for SigNet and RegTest?

src/chainparams.cpp # Should we set 2.5 min nPowTargetSpacing for SigNet adn RegTest?

## Scheduler related questions ##
src/test/util/setup_common.cpp # Why was *m_node.scheduler, removed from line 202?

src/test/denialofservice_tests.cpp # Why was *m_node.scheduler removed from peerman call?

src/init.cpp # Why was *m_node.scheduler removed from peerman call? Explain to me how the node peer scheduler works?

File: src/chainparams.cpp [hunk 1]
```diff
@@ -64,21 +64,26 @@ class CMainParams : public CChainParams {
...
-        consensus.BIP65Height = 388381; // 000000000000000004c2b624ed5d7756c508d90fd0da2c7c679febfa6c4735f0
-        consensus.BIP66Height = 363725; // 00000000000000000379eaa19dce8c9b722d46ae6a57c2f1a988119488b50931
-        consensus.CSVHeight = 419328; // 000000000000000004a1b34462cb8aeebd5799177f7a29cf28f2d1961716b5b5
-        consensus.SegwitHeight = 481824; // 0000000000000000001c8018d9cb3b742ef25114f27563e3fc4a1902167f9893
-        consensus.MinBIP9WarningHeight = 483840; // segwit activation height + miner confirmation window
-        consensus.powLimit = uint256S("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
-        consensus.nPowTargetTimespan = 14 * 24 * 60 * 60; // two weeks
-        consensus.nPowTargetSpacing = 10 * 60;
+        consensus.nSubsidyHalvingInterval = 840000;
+        consensus.BIP16Exception = uint256{};
```
Comment (cruelnovo) [inline src/chainparams.cpp:68]: src/chainparams.cpp  # potential syntax error: `consensus.BIP16Exception = uint256{};` should be: `consensus.BIP16Exception = uint256();`

File: src/policy/fees.cpp [hunk 2]
```diff
@@ -562,7 +562,7 @@ void CBlockPolicyEstimator::processTransaction(const CTxMemPoolEntry& entry, boo
     }
     trackedTxs++;
 
-    // Feerates are stored and reported as BTC-per-kb:
+    // Feerates are stored and reported as VTC-per-kb:
```
Comment (cruelnovo) [inline src/policy/fees.cpp:565]: Line 595: update comment to VTC-per-kb

File: src/Makefile.am [hunk 3]
```diff
@@ -657,91 +685,91 @@ bitcoin_bin_ldadd = \
...
-bitcoin_node_SOURCES = $(bitcoin_daemon_sources) init/bitcoin-node.cpp
-bitcoin_node_CPPFLAGS = $(bitcoin_bin_cppflags)
-bitcoin_node_CXXFLAGS = $(bitcoin_bin_cxxflags)
-bitcoin_node_LDFLAGS = $(bitcoin_bin_ldflags)
-bitcoin_node_LDADD = $(LIBBITCOIN_SERVER) $(bitcoin_bin_ldadd) $(LIBBITCOIN_IPC) $(LIBMULTIPROCESS_LIBS)
+vertcoin_node_SOURCES = $(vertcoin_daemon_sources) init/bitcoin-node.cpp
+vertcoin_node_CPPFLAGS = $(vertcoin_bin_cppflags)
+vertcoin_node_CXXFLAGS = $(vertcoin_bin_cxxflags)
+vertcoin_node_LDFLAGS = $(vertcoin_bin_ldflags)
+vertcoin_node_LDADD = $(LIBBITCOIN_SERVER) $(bitcoin_bin_ldadd) $(LIBBITCOIN_IPC) $(LIBMULTIPROCESS_LIBS)
```
Comment (cruelnovo) [inline src/Makefile.am]:
```suggestion
vertcoin_node_LDADD = $(LIBBITCOIN_SERVER) $(vertcoin_bin_ldadd) $(LIBBITCOIN_IPC) $(LIBMULTIPROCESS_LIBS)
```

File: src/qt/bitcoinstrings.cpp [hunk 4]
```diff
@@ -8,234 +8,234 @@
 #else
 #define UNUSED
 #endif
-static const char UNUSED *bitcoin_strings[] = {
-QT_TRANSLATE_NOOP("bitcoin-core", "The %s developers"),
-QT_TRANSLATE_NOOP("bitcoin-core", ""
-"%s corrupt. Try using the wallet tool bitcoin-wallet to salvage or restoring "
+static const char UNUSED *vertcoin_strings[] = {
```
Comment (cruelnovo) [inline src/qt/bitcoinstrings.cpp]:
```suggestion
static const char UNUSED *bitcoin_strings[] = {
```

File: src/qt/bitcoinstrings.cpp [hunk 5]
```diff
@@ -8,234 +8,234 @@
...
+QT_TRANSLATE_NOOP("vertcoin-core", "The %s developers"),
```
Comment (cruelnovo) [inline src/qt/bitcoinstrings.cpp:12]:
```suggestion
QT_TRANSLATE_NOOP("vertcoin-core", "Vertcoin Core"),
QT_TRANSLATE_NOOP("vertcoin-core", "The %s developers"),
```

File: src/qt/forms/receiverequestdialog.ui [hunk 6]
```diff
@@ -65,7 +65,7 @@
    <item row="2" column="1" alignment="Qt::AlignTop">
     <widget class="QLabel" name="uri_content">
      <property name="text">
-      <string notr="true">bitcoin:BC1…</string>
+      <string notr="true">vertcoin:VTC1…</string>
      </property>
      <property name="textFormat">
       <enum>Qt::RichText</enum>
```
Comment (cruelnovo) [inline src/qt/forms/receiverequestdialog.ui:71]: At Line 129: replace BTC with VTC

**Review by cruelnovo — COMMENTED** [review]
A few of my review change notes are remaining on the PR, I could not comment on an appropriate line (line not part of the commit).

**Comment by cruelnovo** [comment]
contrib/guix/guix-* shell scripts need to be executable

**Comment by cruelnovo** [comment]
depends/gen_id shell script needs to be executable

File: src/qt/forms/receiverequestdialog.ui (hunk 6 above)
Comment (sylvesternumski) [reply src/qt/forms/receiverequestdialog.ui:71]: Not sure if it matters here, but you replaced BC with VTC.

File: src/policy/fees.cpp (hunk 2 above)
Comment (vertiond) [reply src/policy/fees.cpp:565]: I think we can leave comments and anything that is not user facing.

File: src/qt/bitcoinstrings.cpp (hunk 5 above)
Comment (vertiond) [reply src/qt/bitcoinstrings.cpp:12]: What is the purpose of this?

File: src/qt/forms/receiverequestdialog.ui (hunk 6 above)
Comment (vertiond) [reply src/qt/forms/receiverequestdialog.ui:71]: Yeah, our bech32 address prefix is VTC whereas upstream's is BC

**Comment by vertiond** [comment]
> contrib/devtools/gen-manpages.sh # Can we generate Vertcoin branded manpages?

Sure, I don't see why not.

> contrib/seeds/makeseeds.py # Slight differences between v0.18 & v19

Old comment is removed https://github.com/vertcoin-project/vertcoin-core/blob/0.18.0/contrib/seeds/makeseeds.py#L30

> src/test/netbase_tests.cpp # Looks like the ports were not updated for Vertcoin

Okay, will update these.

> src/Makefile.am # How come bignum.h was removed?

bignum.h is included https://github.com/vertcoin-project/vertcoin-core/blob/19.0/src/Makefile.am#L124

**Comment by vertiond** [comment]
> doc/man/ # Since we're renaming all the manpage files to 'vertcoin-X.1', should we also update the contents to be vertcoin branded?

Will be covered when we generate Vertcoin branded manpages.

> src/qt/guiutil.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments
src/qt/intro.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

I think we can leave comments and anything that is not user facing.

> src/qt/sendcoinsdialog.cpp # Missed some Vertcoin branding and PSVT vs PSBT crops up again

Got it. As for PSBT, I'm opposed to adjusting the `PSBT` acronym due to it being globally recognized as a standard transaction format.  I just checked source codes following upstream 0.21 from LTC and GRS which continue to use `PSBT`.

> src/qt/forms/receiverequestdialog.ui # Missed an update for vtc1 and VTC string
Where? @cruelnovo 

> src/qt/bitcoingui.cpp # PSBT vs PSVT - 0.18 has it PSVT (Partially Sent Vertcoin Transaction)

Check comment above referring to global recognition of acronym 'PSBT'

> src/rpc/misc.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

Let's leave comments alone.

> src/wallet/test/wallet_tests.cpp # Update strings from bitcoin to vertcoin
Got it.

> src/bitcoin-cli.cpp # Update for a number of strings from bitcoin to vertcoin

Got it.

> src/bitcoind.cpp # Update for a number of strings from bitcoin to vertcoin

Got it. Leaving comments alone.

**Comment by vertiond** [comment]
> src/util/system.cpp # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments. How is the CopyrightHolders function working? Why the strPrefix2?

The strPrefix2 is to allow two copyright holders (Vertcoin Developers and Bitcoin Developers) to appear in places like the opening splash screen.

> src/policy/fees.cpp # only a difference in a comment. BTC-per-kb vs VTC-per-kb
src/Makefile.qt.include # Vertcoin Core 0.18 changed bitcoin to vertcoin in comments

Leaving comments alone.

**Comment by vertiond** [comment]
> src/crypto/verthash.cpp # How did you know to use gArgs.GetDataDirNet() instead of GetDataDir() (0.18 uses GetDataDir())
src/crypto/verthash_datafile.cpp # How did you know to use gArgs.GetDataDirNet() instead of GetDataDir() (0.18 uses GetDataDir())

There was a compiling error without and I adjusted to how upstream manages it now.

> contrib/seeds/nodes_test.txt # can we provide a list of known testnet seeds? or remove this file?

Making a list of testnet seeds would be a good idea. @cruelnovo could generate such a list from their seeder.

> contrib/verifybinaries/ # Should we remove this directory from Vertcoin?

Looks like this hasn't been updated post-guix on upstream and is awaiting to be refactored or removed. It doesn't work anyway but let's leave it. We will instead make the verification process as clear as possible in each release.

> src/bignum.h # is this part of KimotoGravityWell? It is used by pow.cpp

Yes, https://github.com/vertcoin-project/vertcoin-core/commit/79e8a0b227e842ba604b2b4f64107ed6f3c0fb2e

> src/chainparamsseeds.h # How come the seeds are not needed for Vertcoin in BIP155 format?

That's just how the contrib/seeds/generate-seeds.py script works now.

> src/pow.cpp # How did you know to fix the "blockstogoback" logic (line 87-94)

This was in Vertcoin historically and is a remnant from Litecoin https://github.com/litecoin-project/litecoin/commit/b1b31d15cc720a1c186431b21ecc9d1a9062bcb6

> configure.ac # how come we're no longer enabling the SSE2 compiler extension?

It was never enabled by default and has been broken until 0.18.0.  It's also unclear if it works properly given there is no test written for it. Scrypt-N blocks verify quickly so I don't see it as a huge issue. This can always be looked at and added later by someone interested. Related issue is https://github.com/vertcoin-project/vertcoin-core/issues/177

> src/chainparams.cpp # Should we set nPowTargetTimespan to 3.5 days for SigNet and RegTest?
src/chainparams.cpp # Should we set 2.5 min nPowTargetSpacing for SigNet adn RegTest?

That would be a target spacing for KGW of which only mainnet uses.

**Comment by vertiond** [comment]
>src/test/util/setup_common.cpp # Why was *m_node.scheduler, removed from line 202?
src/test/denialofservice_tests.cpp # Why was *m_node.scheduler removed from peerman call?
src/init.cpp # Why was *m_node.scheduler removed from peerman call? Explain to me how the node peer scheduler works?

Related commit: https://github.com/vertcoin-project/vertcoin-core/commit/1ababdd4c588d1f639063d9acc280bfc9823e192

Without this commit, startup fails when having to create verthash. I'm not sure as to why but I know it has something to do with the order in which verthash is generated in src/init.cpp. This commit is backported from upstream 23.0.

File: src/qt/forms/receiverequestdialog.ui (hunk 6 above)
Comment (cruelnovo) [inline src/qt/forms/receiverequestdialog.ui:71]: Line 100: <string notr="true">bc1…</string>
Update to:
<string notr="true">vtc1…</string>

File: src/qt/bitcoinstrings.cpp (hunk 5 above)
Comment (vertiond) [reply src/qt/bitcoinstrings.cpp:12]: Seems that bitcoinstrings.cpp needs no modifying for vertcoin and is only necessary for the translation process of which we have left entirely to upstream.

**Review by cruelnovo — APPROVED** [review]
Backports from upstream 22.1 are good

**Review by cruelnovo — APPROVED** [review]
End-user branding for Vertcoin Core is good.  Linking flags for 'vertcoin_node_LDADD' are set.

**Pull request merged**